storage/myisam/mi_check.c

```diff
@@ -3430,8 +3430,8 @@ static int sort_get_next_record(MI_SORT_PARAM *sort_param)
 	  {
 	    if (!searching)
 	      mi_check_print_info(param,
-				  "Found block with impossible length %u at %s; Skipped",
-				  block_info.block_len+ (uint) (block_info.filepos-pos),
+				  "Found block with impossible length %lu at %s; Skipped",
+				  block_info.block_len+ (ulong) (block_info.filepos-pos),
```

Comment:
I forgot how I came up with this…

Other `(block_info.filepos-pos)`s nearby were casted to `(uint)`s.
For consistency, either this should be reverted or the other instances should also change (improve).